service/src/main/java/bio/terra/user/db/ProfileDao.java

```diff
@@ -85,7 +86,7 @@ private String getPropertySingle(String userId, List<String> path) {
             return rs.getString("value");
           });
     } catch (EmptyResultDataAccessException e) {
-      return "{}";
+      return CollectionUtils.isEmpty(path) ? "{}" : null;
```

Comment:
could you explain why returning {} when path is empty?

also annotate the method and getProperty with @nullable now you are retuning null in some cases.

Reply:
This combined with not being able to directly set the root means that clients can expect the profile object to always exist.

Reply:
oh i see. given you create a row when non exist, does it still throw emptyReuslt exception when path is empty? I thought it would just return {}

Reply:
Atm the row create is only attempted on set so this exception is possible when trying to read for the first time. 

Do you think it should also be created on get? I did it like this to save on queries, but one basically no-op query shouldn't really matter so...

Reply:
i see. thanks for explaining. please add a comment